### PR TITLE
[Node 4] Upgrade to NAN 2.x for Node 4 compatibility

### LINKED ISSUE #9: Update to nan 2 for Node.js 4.0.0 compatibility.
Adding to the list: https://github.com/nodejs/node/issues/2798.
Related: #8.

### PR DESCRIPTION
**Pull request by ide**
`npm install` succeeds. Could you let me know how to test this thoroughly?

Fixes #9.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -24,7 +27,7 @@
     "Ben Trask <bentrask@comcast.net>"
   ],
   "engines": {
-    "node": ">0.7.9"
+    "node": ">=4"
```
Comment (ChALkeR) [inline package.json]: What for? Won't it work with 0.12?

Comment (ide) [inline package.json]: Maybe it will. I was mostly interested in supporting current Node.

**Comment by KenanSulayman** [comment]
We need this for libpcap in a realtime network monitor, please merge as soon as possible.

Thanks!

**Comment by btrask** [comment]
I've given @ide push access to this repo.

**Comment by KenanSulayman** [comment]
:+1: Thanks!

**Comment by ide** [comment]
Thanks :D

**Comment by btrask** [comment]
Someone with more interest and familiarity with npm should take over doing the releases. I can give consent to the npm people if needed.

**Comment by ide** [comment]
@btrask I'm familiar with npm but am not the right person to maintain this project (just using it as a hobbyist). I'd be happy to do the next npm release though if you want to add me as a publisher (`npm owner add ide`).

**Comment by btrask** [comment]
Done.

Thanks for doing it. Someone get this guy a corporate sponsorship.

**Comment by ide** [comment]
I published master as 0.3.0. Did a version bump because I wasn't confident that this would work with older versions of Node and wanted users of the package to be more aware of the change.

**Pull request merged**